### PR TITLE
[ MB-14478 ] Move history shows creation of MTO agent in move history table and hides deletion when the shipment is deleted

### PR DESCRIPTION
**Pull request by mmayo-truss**
## Summary

You can create an MTO agent (receiving or releasing) in two ways with distinct eventNames. If you fill out the agent information during the shipment creation, it will have an event name of createShipment. If you go back to edit the shipment and add an agent, it will have an event name of updateShipment. Currently, only the updateShipment events return data in the move history rows. This PR remedies that.

Also, when you delete a shipment any agents get updated to include a deleted at timestamp. These updates are showing up as unnecessary rows surrounding the deleted shipment row. This PR remedies this as well.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the milmove app as a service member
2. Create a move
3. When you get to the shipment creation, fill out only one of the agents, either releasing or receiving
4. Before you submit the move go back and edit the shipment. Update the information on the agent you already filled out and add the other type of agent
5. Sign and submit the move
6. Log in as a services councilor and view the move history table for the move you submitted
7. You should see entries for the creations of and updates to the mto_agents in the move history table
8. Go and edit the shipment as a service councilor and delete the shipment. 
9. You should see no accompanying rows for the deletion of the agents. You should only see the deleted shipment row.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Screenshots
Before
image

After
image

### CODE REVIEW

**Review by juliemoon — APPROVED** [review]
Screen Shot 2022-11-22 at 10 31 46 AM
@mmayo-truss thanks, I'm able to see all the changes. This lgtm!

File: src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/createMTOShipmentAgent.jsx [hunk 1]
```diff
@@ -2,37 +2,24 @@ import React from 'react';
...
   const newChangedValues = {
-    shipment_type: context[0]?.shipment_type,
-    shipment_id_display: context[0]?.shipment_id_abbr.toUpperCase(),
-    ...changedValues,
+    ...formatMoveHistoryAgent(changedValues),
+    ...getMtoShipmentLabel(historyRecord),
   };
 
-  newChangedValues[agentLabel] = agent;
-
```
Comment (ruizajtruss) [inline src/constants/MoveHistory/EventTemplates/UpdateMTOShipment/createMTOShipmentAgent.jsx:29]: **Praise:** Love how much less code this is now 🎉

File: src/utils/formatters.js [hunk 2]
```diff
@@ -236,7 +243,10 @@ export const formatMoveHistoryAgent = (agent) => {
 
   formattedAgent = formattedAgent.trim();
 
-  return formattedAgent;
+  const formattedAgentValues = {};
+  formattedAgentValues[agentLabel] = formattedAgent;
+
+  return formattedAgentValues;
```
Comment (ruizajtruss) [inline src/utils/formatters.js:249]: **Praise:** Great move adding this in, vastly improves how we write out our templates

**Review by ruizajtruss — APPROVED** [review]
Everything looks good, great work on this! 🎉

**Pull request merged**